packages/aws-cdk-lib/aws-lambda/lib/function.ts

```diff
@@ -1497,10 +1510,17 @@ Environment variables can be marked for removal when used in Lambda@Edge by sett
     // List can't be empty here, if we got this far you intended to put your Lambda
     // in subnets. We're going to guarantee that we get the nice error message by
     // making VpcNetwork do the selection again.
-
-    return {
-      subnetIds: selectedSubnets.subnetIds,
-      securityGroupIds: securityGroups.map(sg => sg.securityGroupId),
+    if (props.ipv6AllowedForDualStack) {
+      return {
+        ipv6AllowedForDualStack: ipv6AllowedForDualStack,
```

Comment:
What happens if we return `ipv6AllowedForDualStack: false`? Just thinking if we need this if statement.

Reply:
The default value of `ipvAllowedForDualStack` is false, so even if we don't set it to the construct, it'll stay false. We have made the decision of adding `ipv6AllowedForDualStack` to return config only if it is set to true, because when it is not set, it was breaking other integration test, where VpcConfig is being used. As this property is set for dualStack VPC only and the other test might not be using dualStack VPC.